lib/util/ast.js

```diff
@@ -50,7 +45,7 @@ function isHookCall(node) {
 function findReference(scope, node) {
     const hasSameRangeAsNode = pathEq([ 'identifier', 'range' ], node.range);
 
-    return find(hasSameRangeAsNode, scope.references);
+    return find(() => hasSameRangeAsNode, scope.references);
```

Comment:
This looks incorrect. The value of `hasSameRangeAsNode` should be a function, which accepts a `reference` object and compares the value of its `identifier.range` path with the value given in `node.value`. But now it is wrapped in another function, ignoring any input value and always returning the function, which is a truethy value.

I wonder why no test is failing.

Reply:
I’ve implemented two unit tests here https://github.com/lo1tuma/eslint-plugin-mocha/pull/328 which should fail when `findReference` is not working correctly.

Reply:
Yeah i think i did that because vscode/ts was linting that it was incorrect, but the original implementation was correct, just TS not understanding that pathEq is the correct type
Luckily this doesn't use TSLint so this isn't a problem 🙃